tedana/decomposition/pca.py

```diff
@@ -367,6 +367,13 @@ def tedpca(
         metrics=required_metrics,
     )
 
+    # Flip signs for voxel_comp_weights to match comp_ts mixing matrix
+    # that were flipped in collect.generate_metrics
+    optimal_sign = np.matlib.repmat(
+        component_table["optimal sign"].to_numpy(), voxel_comp_weights.shape[0], 1
+    )
+    voxel_comp_weights = voxel_comp_weights * optimal_sign
```

Comment:
I don't think we need a full matrix for the signs.

```suggestion
    optimal_sign = component_table["optimal sign"].to_numpy()
    voxel_comp_weights = voxel_comp_weights * optimal_sign[:, None]
```

Reply:
Ahhh that makes sense. Sorry!